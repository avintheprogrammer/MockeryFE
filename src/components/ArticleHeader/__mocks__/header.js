// import wireLogo from '../../../../assets/images/fpo/sponsor-logo.svg';
import wireLogo from '../../../../assets/images/fpo/wirestory-logo.jpg';
import authorImage from '../../../../assets/images/fpo/deidrebosa.jpg';
import featuredImage from '../../../../assets/images/fpo/featuredimage.jpg';
import boardroomImage from '../../../../assets/images/fpo/boardroom.jpg';
import accentureLogo from '../../../../assets/images/fpo/accenture-consulting-logo.jpg';

const mocks = {
  id: 104314759,
  type: 'mockerynewsstory',
  native: false,
  premium: false,
  socialtoolsEnabled: true,
  datePublished: '2017-03-02T06:16:15 GMT',
  dateModified: '2018-02-07T03:19:16 GMT',
  datePublishedFormatted: '2017-03-02T06:16:15 GMT',
  dateLastPublishedFormatted: '2018-02-07T03:19:16 GMT',
  description:
    "Casino operators globally are eyeing entry into Japan, but one analyst said Japanese lawmakers may adopt Singapore's methods in limiting gambling. ",
  headline: 'Japan may pull a Singapore in controlling casino gambling: Analyst',
  title: 'Japan may pull a Singapore in controlling casino gambling: Analyst',
  sourceOrganization: [
    {
      id: '80000008',
      url: null,
      name: 'Reuters',
      logo:
        'https://qa-aws01fm.mockery.com/applications/mockery.com/resources/source_logo/2018/01/26/132195270-reuters.gif',
    },
  ],
  author: [
    {
      id: 103944451,
      name: 'Bill Smith',
      url: 'http://qa-pa13pub.mockery.com/testcreator/',
      image: {
        url: null,
      },
      sameAs: null,
      socialMediaInfo: null,
    },
  ],
  promoImage: {
    url: featuredImage,
    caption: 'Image courtesy of Getty',
  },
  featuredMedia: {
    id: 100518123,
    type: null,
    caption: 'Las Vegas Sands Corp, The Venetian.',
    copyrightHolder: 'Ricky Barnard | Wikipedia',
    representativeOfPage: true,
    url: 'http://img-static-qa1.mockery.com/image/104769012-SLSR_Dallas_exterior.jpg',
    datePublished: '2013-03-04T14:38:53+0000',
  },
  publisher: {
    name: 'MOCKERY',
    logo: 'http://sc.mockeryfm.com/applications/mockery.com/staticcontent/img/mockery-hdr-logo2.png',
  },
  section: {
    id: 10001165,
    color: '#1FF1C8',
    name: 'Casinos and Gaming',
    url: 'http://qa-pa13pub.mockery.com/id/10001165',
    image:
      'https://fm.mockery.com/applications/mockery.com/resources/styles/skin/special-reports/the-pulse/pulse-1market-header-2800.jpg',
    logo:
      'https://fm.mockery.com/applications/mockery.com/resources/img/editorial/2017/Redesign/pulse_logo.png',
  },
};

const noImageHeader = {
  id: 104314759,
  type: 'mockerynewsstory',
  native: false,
  premium: false,
  socialtoolsEnabled: true,
  datePublished: '2017-03-02T06:16:15 GMT',
  dateModified: '2018-02-07T03:19:16 GMT',
  datePublishedFormatted: '2017-03-02T06:16:15 GMT',
  dateLastPublishedFormatted: '2018-02-07T03:19:16 GMT',
  description:
    "Casino operators globally are eyeing entry into Japan, but one analyst said Japanese lawmakers may adopt Singapore's methods in limiting gambling. ",
  headline: 'Japan may pull a Singapore in controlling casino gambling: Analyst',
  title: 'Japan may pull a Singapore in controlling casino gambling: Analyst',
  sourceOrganization: [
    {
      id: '80000008',
      url: null,
      name: 'Reuters',
      logo: wireLogo,
    },
  ],
  author: [
    {
      id: 103944451,
      name: 'Bill Smith',
      url: 'http://qa-pa13pub.mockery.com/testcreator/',
      image: {
        url: null,
      },
      sameAs: null,
      socialMediaInfo: [
        {
          type: 'twitter',
          url: 'http://twitter.com',
          displayText: '@helloooo',
        },
      ],
    },
  ],
  promoImage: null,
  featuredMedia: null,
  publisher: {
    name: 'MOCKERY',
    logo: 'http://sc.mockeryfm.com/applications/mockery.com/staticcontent/img/mockery-hdr-logo2.png',
  },
  section: {
    id: 10001165,
    color: '#1FF1C8',
    name: 'Casinos and Gaming',
    url: 'http://qa-pa13pub.mockery.com/id/10001165',
    image:
      'https://fm.mockery.com/applications/mockery.com/resources/styles/skin/special-reports/the-pulse/pulse-1market-header-2800.jpg',
    logo:
      'https://fm.mockery.com/applications/mockery.com/resources/img/editorial/2017/Redesign/pulse_logo.png',
  },
};

const noImageHeaderTwoAuthors = {
  id: 104314759,
  type: 'mockerynewsstory',
  datePublished: '2017-03-02T06:16:15 GMT',
  dateModified: '2018-02-07T03:19:16 GMT',
  datePublishedFormatted: '2017-03-02T06:16:15 GMT',
  dateLastPublishedFormatted: '2018-02-07T03:19:16 GMT',
  description:
    "Casino operators globally are eyeing entry into Japan, but one analyst said Japanese lawmakers may adopt Singapore's methods in limiting gambling. ",
  headline: 'Japan may pull a Singapore in controlling casino gambling: Analyst',
  title: 'Japan may pull a Singapore in controlling casino gambling: Analyst',
  sourceOrganization: [
    {
      id: '80000008',
      url: null,
      name: 'Reuters',
      logo: wireLogo,
    },
  ],
  author: [
    {
      id: 104799799,
      name: 'Sharon L. Lynch',
      url: 'http://qa-pa14pub.mockery.com/sharon-l-lynch/',
      image: { url: authorImage },
      sameAs: null,
      socialMediaInfo: [
        {
          url: 'https://twitter.com/sharonlynch',
          displayText: 'sharonlynch',
          type: 'twitter',
        },
      ],
    },
    {
      id: 123456789,
      name: 'Deirdre Bosa',
      url: 'http://qa-pa14pub.mockery.com/sharon-l-lynch/',
      image: { url: authorImage },
      sameAs: null,
      socialMediaInfo: [
        {
          url: 'https://twitter.com/deirdebosa',
          displayText: 'deirdebosa',
          type: 'twitter',
        },
      ],
    },
  ],
  promoImage: {
    url: featuredImage,
    caption: 'Image courtesy of Getty',
  },
  featuredMedia: null,
  publisher: {
    name: 'MOCKERY',
    logo: 'http://sc.mockeryfm.com/applications/mockery.com/staticcontent/img/mockery-hdr-logo2.png',
  },
  section: {
    id: 10001165,
    color: '#1FF1C8',
    name: 'Casinos and Gaming',
    url: 'http://qa-pa13pub.mockery.com/id/10001165',
    image:
      'https://fm.mockery.com/applications/mockery.com/resources/styles/skin/special-reports/the-pulse/pulse-1market-header-2800.jpg',
    logo:
      'https://fm.mockery.com/applications/mockery.com/resources/img/editorial/2017/Redesign/pulse_logo.png',
  },
};

const noImageHeaderWithWire = {
  id: 104314759,
  type: 'mockerynewsstory',
  datePublished: '2017-03-02T06:16:15 GMT',
  dateModified: '2018-02-07T03:19:16 GMT',
  datePublishedFormatted: '2017-03-02T06:16:15 GMT',
  dateLastPublishedFormatted: '2018-02-07T03:19:16 GMT',
  description:
    "Casino operators globally are eyeing entry into Japan, but one analyst said Japanese lawmakers may adopt Singapore's methods in limiting gambling. ",
  headline: 'Japan may pull a Singapore in controlling casino gambling: Analyst',
  title: 'Japan may pull a Singapore in controlling casino gambling: Analyst',
  sourceOrganization: [
    {
      id: '80000008',
      url: null,
      name: 'Reuters',
      logo: wireLogo,
    },
  ],
  author: [
    {
      id: 104799799,
      name: 'Sharon L. Lynch',
      url: 'http://qa-pa14pub.mockery.com/sharon-l-lynch/',
      image: { url: authorImage },
      sameAs: null,
      socialMediaInfo: [
        {
          url: 'https://twitter.com/sharonlynch',
          displayText: 'sharonlynch',
          type: 'twitter',
        },
      ],
    },
  ],
  promoImage: null,
  featuredMedia: null,
  publisher: {
    name: 'REUTERS',
    logo: wireLogo,
  },
  section: {
    id: 10001165,
    color: '#1FF1C8',
    name: 'Casinos and Gaming',
    url: 'http://qa-pa13pub.mockery.com/id/10001165',
    image:
      'https://fm.mockery.com/applications/mockery.com/resources/styles/skin/special-reports/the-pulse/pulse-1market-header-2800.jpg',
    logo:
      'https://fm.mockery.com/applications/mockery.com/resources/img/editorial/2017/Redesign/pulse_logo.png',
  },
};

const imageHeader = {
  id: 104314759,
  type: 'mockerynewsstory',
  native: false,
  premium: false,
  socialtoolsEnabled: true,
  datePublished: '2017-03-02T06:16:15 GMT',
  dateModified: '2018-02-07T03:19:16 GMT',
  datePublishedFormatted: '2017-03-02T06:16:15 GMT',
  dateLastPublishedFormatted: '2018-02-07T03:19:16 GMT',
  description:
    "Casino operators globally are eyeing entry into Japan, but one analyst said Japanese lawmakers may adopt Singapore's methods in limiting gambling. ",
  headline: 'Japan may pull a Singapore in controlling casino gambling: Analyst',
  title: 'Japan may pull a Singapore in controlling casino gambling: Analyst',
  sourceOrganization: [
    {
      id: '80000008',
      url: null,
      name: 'Reuters',
      logo: wireLogo,
    },
  ],
  author: [
    {
      id: 103944451,
      name: 'Bill Smith',
      url: 'http://qa-pa13pub.mockery.com/testcreator/',
      image: {
        url: null,
      },
      sameAs: null,
      socialMediaInfo: [
        {
          url: 'http://twitter.com',
          type: 'twitter',
          displayText: '@billsmith',
        },
      ],
    },
  ],
  promoImage: {
    url: featuredImage,
    caption: 'Image courtesy of Getty',
  },
  featuredMedia: {
    id: 100518123,
    type: null,
    caption: 'Las Vegas Sands Corp, The Venetian.',
    copyrightHolder: 'Ricky Barnard | Wikipedia',
    representativeOfPage: true,
    url: 'http://img-static-qa1.mockery.com/image/104769012-SLSR_Dallas_exterior.jpg',
    datePublished: '2013-03-04T14:38:53+0000',
  },
  publisher: null,
  section: {
    id: 10001165,
    color: '#1FF1C8',
    name: 'Casinos and Gaming',
    url: 'http://qa-pa13pub.mockery.com/id/10001165',
    image: boardroomImage,
    logo:
      'https://fm.mockery.com/applications/mockery.com/resources/img/editorial/2017/Redesign/pulse_logo.png',
  },
};

const imageHeaderFullAuthor = {
  id: 104314759,
  type: 'mockerynewsstory',
  datePublished: '2017-03-02T06:16:15 GMT',
  dateModified: '2018-02-07T03:19:16 GMT',
  datePublishedFormatted: '2017-03-02T06:16:15 GMT',
  dateLastPublishedFormatted: '2018-02-07T03:19:16 GMT',
  description:
    "Casino operators globally are eyeing entry into Japan, but one analyst said Japanese lawmakers may adopt Singapore's methods in limiting gambling. ",
  headline: 'Japan may pull a Singapore in controlling casino gambling: Analyst',
  title: 'Japan may pull a Singapore in controlling casino gambling: Analyst',
  sourceOrganization: [
    {
      id: '80000008',
      url: null,
      name: 'Reuters',
      logo: wireLogo,
    },
  ],
  author: [
    {
      id: 123456789,
      name: 'Deirdre Bosa',
      url: 'http://qa-pa14pub.mockery.com/sharon-l-lynch/',
      image: {
        url:
          'https://fm.mockery.com/applications/mockery.com/resources/img/editorial/2017/08/23/104668777-BOSA_D_9.333x14.240x240.jpg?v=1503511700',
      },
      sameAs: null,
      socialMediaInfo: [
        {
          url: 'https://twitter.com/deirdebosa',
          displayText: 'deirdebosa',
          type: 'twitter',
        },
      ],
    },
  ],
  promoImage: {
    url: featuredImage,
    caption: 'Image courtesy of Getty',
  },
  featuredMedia: {
    id: 100518123,
    type: null,
    caption: 'Las Vegas Sands Corp, The Venetian.',
    copyrightHolder: 'Ricky Barnard | Wikipedia',
    representativeOfPage: true,
    url: 'http://img-static-qa1.mockery.com/image/104769012-SLSR_Dallas_exterior.jpg',
    datePublished: '2013-03-04T14:38:53+0000',
  },
  publisher: {
    name: 'MOCKERY',
    logo: 'http://sc.mockeryfm.com/applications/mockery.com/staticcontent/img/mockery-hdr-logo2.png',
  },
  section: {
    id: 10001165,
    color: '#1FF1C8',
    name: 'Casinos and Gaming',
    url: 'http://qa-pa13pub.mockery.com/id/10001165',
    image: boardroomImage,
    logo:
      'https://fm.mockery.com/applications/mockery.com/resources/img/editorial/2017/Redesign/pulse_logo.png',
  },
};

const videoHeader = {
  id: 104314759,
  type: 'mockerynewsstory',
  native: false,
  premium: false,
  socialtoolsEnabled: true,
  datePublished: '2017-03-02T06:16:15 GMT',
  dateModified: '2018-02-07T03:19:16 GMT',
  datePublishedFormatted: '2017-03-02T06:16:15 GMT',
  dateLastPublishedFormatted: '2018-02-07T03:19:16 GMT',
  description:
    "Casino operators globally are eyeing entry into Japan, but one analyst said Japanese lawmakers may adopt Singapore's methods in limiting gambling. ",
  headline: 'Japan may pull a Singapore in controlling casino gambling: Analyst',
  title: 'Japan may pull a Singapore in controlling casinor gambling: Analyst',
  author: null,
  sourceOrganization: [
    {
      id: '80000008',
      url: null,
      name: 'Reuters',
      logo: wireLogo,
    },
  ],
  featuredMedia: {
    id: 100518123,
    type: null,
    url: 'http://mockery.com/test.mov',
    datePublished: '2013-03-04T14:38:53+0000',
  },
  promoImage: {
    url: featuredImage,
    caption: 'Image courtesy of Getty',
  },
  publisher: {
    name: 'MOCKERY',
    logo: 'http://sc.mockeryfm.com/applications/mockery.com/staticcontent/img/mockery-hdr-logo2.png',
  },
  section: {
    id: 10001165,
    color: '#1FF1C8',
    name: 'Casinos and Gaming',
    url: 'http://qa-pa13pub.mockery.com/id/10001165',
    image:
      'https://fm.mockery.com/applications/mockery.com/resources/styles/skin/special-reports/the-pulse/pulse-1market-header-2800.jpg',
    logo:
      'https://fm.mockery.com/applications/mockery.com/resources/img/editorial/2017/Redesign/pulse_logo.png',
  },
};

const articleMedMockHeader = {
  id: 104314759,
  type: 'mockerynewsstory',
  native: false,
  premium: false,
  socialtoolsEnabled: false,
  datePublished: '2017-03-02T06:16:15 GMT',
  dateModified: '2018-02-07T03:19:16 GMT',
  datePublishedFormatted: '2017-03-02T06:16:15 GMT',
  dateLastPublishedFormatted: '2018-02-07T03:19:16 GMT',
  description:
    "Casino operators globally are eyeing entry into Japan, but one analyst said Japanese lawmakers may adopt Singapore's methods in limiting gambling. ",
  headline: 'Japan may pull a Singapore in controlling casino gambling: Analyst',
  sourceOrganization: [
    {
      id: '80000008',
      url: null,
      name: 'Reuters',
      logo: wireLogo,
    },
  ],
  title: 'Japan may pull a Singapore in controlling casino gambling: Analyst',
  author: [
    {
      id: 103944451,
      name: 'Bill Smith',
      url: 'http://qa-pa13pub.mockery.com/testcreator/',
      image: {
        url: null,
      },
      sameAs: null,
      socialMediaInfo: [
        {
          url: 'http://twitter.com',
          type: 'twitter',
          displayText: '@firstadopter',
        },
      ],
    },
    {
      id: 103944451,
      name: 'Jarold Harkness',
      url: 'http://qa-pa13pub.mockery.com/testcreator/',
      image: {
        url: null,
      },
      sameAs: null,
      socialMediaInfo: [
        {
          url: 'http://twitter.com',
          type: 'twitter',
          displayText: '@jdog',
        },
      ],
    },
  ],
  promoImage: null,
  featuredMedia: {
    id: 100518123,
    type: null,
    caption: 'Las Vegas Sands Corp, The Venetian.',
    copyrightHolder: 'Ricky Barnard | Wikipedia',
    representativeOfPage: true,
    url: 'http://img-static-qa1.mockery.com/image/104769012-SLSR_Dallas_exterior.jpg',
    datePublished: '2013-03-04T14:38:53 GMT',
  },
  publisher: null,
  section: {
    id: 10001165,
    color: '#1FF1C8',
    name: 'Casinos and Gaming',
    url: 'http://qa-pa13pub.mockery.com/id/10001165',
    image:
      'https://fm.mockery.com/applications/mockery.com/resources/styles/skin/special-reports/the-pulse/pulse-1market-header-2800.jpg',
    logo:
      'https://fm.mockery.com/applications/mockery.com/resources/img/editorial/2017/Redesign/pulse_logo.png',
  },
};

const articleBrandedMockHeader = {
  id: 104314759,
  type: 'mockerynewsstory',
  native: true,
  premium: false,
  socialtoolsEnabled: true,
  datePublished: '2017-03-02T06:16:15 GMT',
  dateModified: '2018-02-07T03:19:16 GMT',
  datePublishedFormatted: '2017-03-02T06:16:15 GMT',
  dateLastPublishedFormatted: '2018-02-07T03:19:16 GMT',
  sourceOrganization: [
    {
      id: '80000008',
      url: null,
      name: 'Reuters',
      logo: wireLogo,
    },
  ],
  description:
    "Casino operators globally are eyeing entry into Japan, but one analyst said Japanese lawmakers may adopt Singapore's methods in limiting gambling. ",
  headline: 'Japan may pull a Singapore in controlling casino gambling: Analyst',
  title: 'Japan may pull a Singapore in controlling casino gambling: Analyst',
  author: [],
  promoImage: {
    url: featuredImage,
    caption: 'Image courtesy of Getty',
  },
  featuredMedia: {
    id: 100518123,
    type: null,
    caption: 'Las Vegas Sands Corp, The Venetian.',
    copyrightHolder: 'Ricky Barnard | Wikipedia',
    representativeOfPage: true,
    url: 'http://img-static-qa1.mockery.com/image/104769012-SLSR_Dallas_exterior.jpg',
    datePublished: '2013-03-04T14:38:53+0000',
  },
  publisher: {
    name: 'Accenture Consulting',
    logo: accentureLogo,
  },
  section: {
    id: 10001165,
    color: '#1FF1C8',
    name: 'Casinos and Gaming',
    url: 'http://qa-pa13pub.mockery.com/id/10001165',
    image:
      'https://fm.mockery.com/applications/mockery.com/resources/styles/skin/special-reports/the-pulse/pulse-1market-header-2800.jpg',
    logo:
      'https://fm.mockery.com/applications/mockery.com/resources/img/editorial/2017/Redesign/pulse_logo.png',
  },
};

const articleProMockHeader = {
  id: 104314759,
  type: 'mockerynewsstory',
  native: false,
  premium: true,
  datePublished: '2017-03-02T06:16:15 GMT',
  dateModified: '2018-02-07T03:19:16 GMT',
  datePublishedFormatted: '2017-03-02T06:16:15 GMT',
  dateLastPublishedFormatted: '2018-02-07T03:19:16 GMT',
  sourceOrganization: [
    {
      id: '80000008',
      url: null,
      name: 'Reuters',
      logo: wireLogo,
    },
  ],
  description:
    "Casino operators globally are eyeing entry into Japan, but one analyst said Japanese lawmakers may adopt Singapore's methods in limiting gambling. ",
  headline: 'Japan may pull a Singapore in controlling casino gambling: Analyst',
  title: 'Japan may pull a Singapore in controlling casino gambling: Analyst',
  author: [
    {
      id: 103944451,
      name: 'Bill Smith',
      url: 'http://qa-pa13pub.mockery.com/testcreator/',
      image: {
        url: null,
      },
      sameAs: null,
      socialMediaInfo: [
        {
          url: 'http://twitter.com',
          type: 'twitter',
          displayText: '@firstadopter',
        },
      ],
    },
  ],
  promoImage: null,
  featuredMedia: {
    id: 100518123,
    type: null,
    caption: 'Las Vegas Sands Corp, The Venetian.',
    copyrightHolder: 'Ricky Barnard | Wikipedia',
    representativeOfPage: true,
    url: 'http://img-static-qa1.mockery.com/image/104769012-SLSR_Dallas_exterior.jpg',
    datePublished: '2013-03-04T14:38:53+0000',
  },
  publisher: null,
  section: {
    id: 10001165,
    color: '#1FF1C8',
    name: 'Casinos and Gaming',
    url: 'http://qa-pa13pub.mockery.com/id/10001165',
    image:
      'https://fm.mockery.com/applications/mockery.com/resources/styles/skin/special-reports/the-pulse/pulse-1market-header-2800.jpg',
    logo:
      'https://fm.mockery.com/applications/mockery.com/resources/img/editorial/2017/Redesign/pulse_logo.png',
  },
};

const creatorOverwriteAsAuthor = {
  id: 104314759,
  type: 'mockerynewsstory',
  datePublished: '2017-03-02T06:16:15 GMT',
  dateModified: '2018-02-07T03:19:16 GMT',
  datePublishedFormatted: '2017-03-02T06:16:15 GMT',
  dateLastPublishedFormatted: '2018-02-07T03:19:16 GMT',
  description:
    "Casino operators globally are eyeing entry into Japan, but one analyst said Japanese lawmakers may adopt Singapore's methods in limiting gambling. ",
  headline: 'Japan may pull a Singapore in controlling casino gambling: Analyst',
  title: 'Japan may pull a Singapore in controlling casino gambling: Analyst',
  creatorOverwrite: 'Thomas Baldwin',
  sourceOrganization: [
    {
      id: '80000008',
      url: null,
      name: 'Reuters',
      logo: wireLogo,
    },
  ],
  author: [
    {
      id: 104799799,
      name: 'Sharon L. Lynch',
      url: 'http://qa-pa14pub.mockery.com/sharon-l-lynch/',
      image: { url: authorImage },
      sameAs: null,
      socialMediaInfo: [
        {
          url: 'https://twitter.com/sharonlynch',
          displayText: 'sharonlynch',
          type: 'twitter',
        },
      ],
    },
    {
      id: 123456789,
      name: 'Deirdre Bosa',
      url: 'http://qa-pa14pub.mockery.com/sharon-l-lynch/',
      image: { url: authorImage },
      sameAs: null,
      socialMediaInfo: [
        {
          url: 'https://twitter.com/deirdebosa',
          displayText: 'deirdebosa',
          type: 'twitter',
        },
      ],
    },
  ],
  promoImage: {
    url: featuredImage,
    caption: 'Image courtesy of Getty',
  },
  featuredMedia: null,
  publisher: {
    name: 'MOCKERY',
    logo: 'http://sc.mockeryfm.com/applications/mockery.com/staticcontent/img/mockery-hdr-logo2.png',
  },
  section: {
    id: 10001165,
    color: '#1FF1C8',
    name: 'Casinos and Gaming',
    url: 'http://qa-pa13pub.mockery.com/id/10001165',
    image:
      'https://fm.mockery.com/applications/mockery.com/resources/styles/skin/special-reports/the-pulse/pulse-1market-header-2800.jpg',
    logo:
      'https://fm.mockery.com/applications/mockery.com/resources/img/editorial/2017/Redesign/pulse_logo.png',
  },
};

export {
  noImageHeader,
  imageHeader,
  videoHeader,
  articleMedMockHeader,
  articleBrandedMockHeader,
  articleProMockHeader,
  mocks,
  imageHeaderFullAuthor,
  noImageHeaderTwoAuthors,
  noImageHeaderWithWire,
  creatorOverwriteAsAuthor,
};

export default mocks;
