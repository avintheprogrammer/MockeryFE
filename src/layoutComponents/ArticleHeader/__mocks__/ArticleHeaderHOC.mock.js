const mock = {
  article: {
    title: 'hello',
    section: null,
    featuredMedia: null,
    native: false,
  },
};

export default mock;
