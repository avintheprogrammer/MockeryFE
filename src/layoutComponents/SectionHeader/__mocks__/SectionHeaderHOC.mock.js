const mock = {
  title: 'hello',
  section: {
    title: 'the section',
  },
};

export default mock;
