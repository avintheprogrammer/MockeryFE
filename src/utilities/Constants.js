export const BUFFETT = 'buffett';
export const MOCKERY = 'mockery';
export const DEAL_OR_NO_DEAL = 'dealornodeal';
export const MAKE_IT = 'makeit';

export const BRANDNAMES = {
  BUFFET: BUFFETT,
  BUFFETT,
  MOCKERY,
  DEAL_OR_NO_DEAL,
  MAKE_IT,
};

export default BRANDNAMES;
