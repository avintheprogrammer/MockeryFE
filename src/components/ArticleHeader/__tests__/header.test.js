import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { MAKE_IT } from 'utilities/Constants';
import { ArticleHeader } from '../index';
import { imageHeader } from '../__mocks__/header';
import {
  imageHeaderHighTouch,
  featuredVideoHeader, } from '../__mocks__/makeitHeader';


configure({ adapter: new Adapter() });

test.skip('renders correctly', () => {
  const tree = shallow(<ArticleHeader data={imageHeader} />);

  expect(toJson(tree)).toMatchSnapshot();
});

test.skip('renders make it header correctly', () => {
  const tree = shallow(<ArticleHeader data={imageHeader} brand={MAKE_IT} />);

  expect(toJson(tree)).toMatchSnapshot();
});

test.skip('renders make it featuredVideoHeader correctly', () => {
  const tree = shallow(<ArticleHeader data={featuredVideoHeader} hasBackground brand={MAKE_IT} />);

  expect(toJson(tree)).toMatchSnapshot();
});

test.skip('renders make it imageHeaderHighTouch correctly', () => {
  const tree = shallow(<ArticleHeader data={imageHeaderHighTouch} hasBackground brand={MAKE_IT} />);

  expect(toJson(tree)).toMatchSnapshot();
});
