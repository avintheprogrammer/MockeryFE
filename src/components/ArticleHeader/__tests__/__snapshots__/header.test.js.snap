// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <header
    className="articleHeader"
    id="main-article-header"
  >
    <div
      className="wrapperHeroNoImage"
    >
      <h1
        className="headline"
      >
        Japan may pull a Singapore in controlling casino gambling: Analyst
      </h1>
      <div
        className="time"
      >
        <time>
          2017-03-02T06:16:15 GMT
        </time>
        <span>
           
          <span
            className="mockeryDivider"
          >
            |
          </span>
           
          <time>
             
            2018-02-07T03:19:16 GMT
          </time>
        </span>
      </div>
    </div>
    <div
      className="authorAndShare"
    >
      <div
        className="authorContainer"
      >
        <div
          className="author"
          key="0"
        >
          <Author
            brand="mockery"
            data={
              Object {
                "id": 103944451,
                "image": Object {
                  "url": null,
                },
                "name": "Bill Smith",
                "sameAs": null,
                "socialMediaInfo": Array [
                  Object {
                    "displayText": "@billsmith",
                    "type": "twitter",
                    "url": "http://twitter.com",
                  },
                ],
                "url": "http://qa-pa13pub.mockery.com/testcreator/",
              }
            }
            dataId={0}
            hideAuthorImages={false}
            publisher={
              Object {
                "id": "80000008",
                "logo": "test-file-stub",
                "name": "Reuters",
                "url": null,
              }
            }
          />
        </div>
      </div>
      <Share
        brand=""
        isDark={false}
        shareText=""
        shareType=""
        title="Japan may pull a Singapore in controlling casino gambling: Analyst"
        url=""
      />
    </div>
  </header>
</div>
`;

exports[`renders make it featuredVideoHeader correctly 1`] = `
<div>
  <header
    className="articleHeader"
    id="main-article-header"
  >
    <LazyLoader
      height="100%"
      transitionType="transition-fade"
    >
      <div
        className="imageHero"
        style={
          Object {
            "backgroundImage": "url(test-file-stub?t=1&w=1340&h=500)",
            "backgroundSize": "100%",
          }
        }
      />
    </LazyLoader>
    <div
      className="heroDescriptionWrapper"
    >
      <div
        className="heroDescription"
      >
        <div
          className="heroDescriptionCaption"
        />
        <div
          className="heroDescriptionCredit"
        />
      </div>
    </div>
    <div
      className="wrapperHero"
    >
      <h1
        className="headline"
      >
        Why this investment company’s ideal hedge fund managers are ‘younger, hungrier’ and ‘volatile’
      </h1>
      <div
        className="time"
      >
        <time>
          Mon, Aug 7, 2017 | 8:41 AM ET
        </time>
        <span>
           
          <span
            className="mockeryDivider"
          >
            |
          </span>
           
          <time>
             
            Updated 8 Hours Ago
          </time>
        </span>
      </div>
    </div>
    <div
      className="authorAndShare"
    >
      <div
        className="authorContainer"
      >
        <div
          className="author"
          key="0"
        >
          <Author
            brand="mockery"
            data={
              Object {
                "id": 104799799,
                "image": "test-file-stubt=1",
                "name": "Sharon L. Lynch",
                "sameAs": null,
                "socialMediaInfo": Array [
                  Object {
                    "displayText": "sharonlynch",
                    "type": "twitter",
                    "url": "https://twitter.com/sharonlynch",
                  },
                ],
                "url": "http://qa-pa14pub.mockery.com/sharon-l-lynch/",
              }
            }
            dataId={0}
            hideAuthorImages={false}
            publisher={null}
          />
        </div>
      </div>
      <Share
        brand=""
        isDark={false}
        shareText=""
        shareType=""
        title="Why this investment company’s ideal hedge fund managers are ‘younger, hungrier’ and ‘volatile’"
        url=""
      />
    </div>
  </header>
</div>
`;

exports[`renders make it header correctly 1`] = `
<div>
  <header
    className="articleHeader"
    id="main-article-header"
  >
    <div
      className="wrapperHeroNoImage"
    >
      <h1
        className="headline"
      >
        Japan may pull a Singapore in controlling casino gambling: Analyst
      </h1>
      <div
        className="time"
      >
        <time>
          2017-03-02T06:16:15 GMT
        </time>
        <span>
           
          <span
            className="mockeryDivider"
          >
            |
          </span>
           
          <time>
             
            2018-02-07T03:19:16 GMT
          </time>
        </span>
      </div>
    </div>
    <div
      className="authorAndShare"
    >
      <div
        className="authorContainer"
      >
        <div
          className="author"
          key="0"
        >
          <Author
            brand="mockery"
            data={
              Object {
                "id": 103944451,
                "image": Object {
                  "url": null,
                },
                "name": "Bill Smith",
                "sameAs": null,
                "socialMediaInfo": Array [
                  Object {
                    "displayText": "@billsmith",
                    "type": "twitter",
                    "url": "http://twitter.com",
                  },
                ],
                "url": "http://qa-pa13pub.mockery.com/testcreator/",
              }
            }
            dataId={0}
            hideAuthorImages={false}
            publisher={
              Object {
                "id": "80000008",
                "logo": "test-file-stub",
                "name": "Reuters",
                "url": null,
              }
            }
          />
        </div>
      </div>
      <Share
        brand=""
        isDark={false}
        shareText=""
        shareType=""
        title="Japan may pull a Singapore in controlling casino gambling: Analyst"
        url=""
      />
    </div>
  </header>
</div>
`;

exports[`renders make it imageHeaderHighTouch correctly 1`] = `
<div>
  <header
    className="articleHeader"
    id="main-article-header"
  >
    <LazyLoader
      height="100%"
      transitionType="transition-fade"
    >
      <div
        className="imageHero"
        style={
          Object {
            "backgroundImage": "url(test-file-stub?t=1&w=1340&h=500)",
            "backgroundSize": "100%",
          }
        }
      />
    </LazyLoader>
    <div
      className="heroDescriptionWrapper"
    >
      <div
        className="heroDescription"
      >
        <div
          className="heroDescriptionCaption"
        />
        <div
          className="heroDescriptionCredit"
        >
          Photo courtesy of Alina Moore
        </div>
      </div>
    </div>
    <div
      className="wrapperHero"
    >
      <h1
        className="headline"
      >
        74 percent of millennials would rather text — here is how to overcome your fear of the phone
      </h1>
      <div
        className="time"
      >
        <time>
          Mon, Aug 7, 2017 | 8:41 AM ET
        </time>
        <span>
           
          <span
            className="mockeryDivider"
          >
            |
          </span>
           
          <time>
             
            Updated 8 Hours Ago
          </time>
        </span>
      </div>
    </div>
    <div
      className="authorAndShare"
    >
      <div
        className="authorContainer"
      >
        <div
          className="author"
          key="0"
        >
          <Author
            brand="mockery"
            data={
              Object {
                "id": 104799799,
                "image": "test-file-stubt=1",
                "name": "Sharon L. Lynch",
                "sameAs": null,
                "socialMediaInfo": Array [
                  Object {
                    "displayText": "sharonlynch",
                    "type": "twitter",
                    "url": "https://twitter.com/sharonlynch",
                  },
                ],
                "url": "http://qa-pa14pub.mockery.com/sharon-l-lynch/",
              }
            }
            dataId={0}
            hideAuthorImages={false}
            publisher={null}
          />
        </div>
      </div>
      <Share
        brand=""
        isDark={false}
        shareText=""
        shareType=""
        title="74 percent of millennials would rather text — here is how to overcome your fear of the phone"
        url=""
      />
    </div>
  </header>
</div>
`;
